src/app/country/country-list/country-list.component.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+import { Subscription, of } from 'rxjs';
+import { Country } from '../../models/country';
+import { CountryService } from '../services/country.service';
+import { CountryServiceMock } from '../../shared/mocks/country-service.mock';
+import { CountryListComponent } from './country-list.component';
+
+describe('CountryListComponent', () => {
+  let component: CountryListComponent;
+  let fixture: ComponentFixture<CountryListComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [CountryListComponent],
+      providers: [{ provide: CountryService, useClass: CountryServiceMock }],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CountryListComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    fixture.detectChanges();
+    expect(component).toBeTruthy();
+  });
+
+  it('should fill countryList variable', () => {
+    expect(component.countryList$).toBeUndefined();
+    const countryService: CountryService = TestBed.inject(CountryService);
+    const getCountryServiceSpy = jest.spyOn(countryService, 'getCountries');
+    fixture.detectChanges();
+    expect(getCountryServiceSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should emit a country', () => {
```

Comment:
This is completely wrong. Here, your test should get the country from the template, click on it, and then, test that the ``emit`` is being done. 

What you're doing is to execute manually the ``emit`` in the test, and then checking that it has been called. That makes no sense.

Reply:
Sure, you are right! I changed it!